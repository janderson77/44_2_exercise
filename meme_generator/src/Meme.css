.Meme-container {
	position: relative;
}

.Meme-container img {
	min-width: 300px;
	min-height: 300px;
	margin: 20px;
	overflow: hidden;
	border-radius: 6px;
}

.Meme-container span {
	display: block;
	font-size: 20px;
	font-weight: bold;
	position: absolute;
	left: 0;
	right: 0;
	margin: auto;
	color: white;
	font-size: 3em;
}

.top-text {
	top: 20px;
	text-shadow: 2px 2px #000000;
	-webkit-text-stroke: 2px black;
}

.bottom-text {
	bottom: 30px;
	text-shadow: 2px 2px #000000;
	-webkit-text-stroke: 2px black;
}
