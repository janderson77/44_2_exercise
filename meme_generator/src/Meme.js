import React from 'react'
import './Meme.css'

const Meme = ({removeMeme, topText, bottomText, url, id}) => {
    const handleRemoveMeme = () => {
        removeMeme(id)
    }
    return (
        <div id={id} className="Meme-container">
            <span className="top-text">{topText}</span>
            <img src={url} alt="meme" className="Meme" />
            <span className="bottom-text">{bottomText}</span>
            <button id="meme_removeBtn" onClick={handleRemoveMeme}>X</button>
        </div>
    )
}

export default Meme