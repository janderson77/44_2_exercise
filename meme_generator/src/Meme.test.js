import React from 'react';
import { render } from '@testing-library/react';
import Meme from './Meme';

let meme;
beforeEach(() => {
    meme = render(
        <Meme 
            url="https://crhscountyline.com/wp-content/uploads/2020/03/Capture.png"
            topText="Wow, such meme"
            bottomText="Much funny. Very wow"
        />
    )
})

it('renders without crashing', () => {
    expect(meme).not.toBeNull();
})

it('matches snapshot', () => {
    expect(meme.asFragment()).toMatchSnapshot()
})

it('puts uses url prop for image src', () => {
    const img = meme.getByAltText("meme");

    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'https://crhscountyline.com/wp-content/uploads/2020/03/Capture.png')
})