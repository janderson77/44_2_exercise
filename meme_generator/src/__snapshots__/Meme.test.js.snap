// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Meme-container"
  >
    <span
      class="top-text"
    >
      Wow, such meme
    </span>
    <img
      alt="meme"
      class="Meme"
      src="https://crhscountyline.com/wp-content/uploads/2020/03/Capture.png"
    />
    <span
      class="bottom-text"
    >
      Much funny. Very wow
    </span>
    <button
      id="meme_removeBtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
